### PR TITLE
BUG: Arithmetic inplace ops not respecting CoW

### PR DESCRIPTION
**Pull request by phofl**
- [ ] xref #49473 (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

We should add tests for all op-possibilities in a follow up

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -11731,10 +11731,16 @@ def _inplace_method(self, other, op):
         """
         result = op(self, other)
 
+        # Item "ArrayManager" of "Union[ArrayManager, SingleArrayManager,
+        # BlockManager, SingleBlockManager]" has no attribute "_has_no_reference"
         if (
             self.ndim == 1
             and result._indexed_same(self)
             and is_dtype_equal(result.dtype, self.dtype)
+            and not (
```
[the remark this answers is not in the document]
Comment (jorisvandenbossche) [reply pandas/core/generic.py:11740]: Yeah, I think it is fine to use the manager method here (we can de-privatize if needed at some point)

[the remark this answers is not in the document]
Comment (jorisvandenbossche) [reply pandas/core/generic.py:11740]: Actually, if you want to use an existing manager method that already takes care of this, on a SingleBlockManager, that would actually be `self._mgr.setitem_inplace(slice(None), result._values)`

That should translate to exactly the same (it only does an additional `np_can_hold_element` check, but I assume that should be cheap if the dtypes are equal, which is already checked above)

[the remark this answers is not in the document]
Comment (phofl) [reply pandas/core/generic.py:11740]: I like this, switched over.

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by phofl** [comment]
greenish

**Pull request merged**